neqo-client/src/main.rs

```diff
@@ -123,135 +121,113 @@ impl KeyUpdateState {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 struct HexArg(Vec<u8>);
-impl FromStr for HexArg {
-    type Err = ClientError;
-
-    fn from_str(s: &str) -> Res<Self> {
-        fn v(c: u8) -> Res<u8> {
-            match c {
-                b'A'..=b'F' => Ok(c - b'A' + 10),
-                b'a'..=b'f' => Ok(c - b'a' + 10),
-                b'0'..=b'9' => Ok(c - b'0'),
-                _ => Err(ClientError::ArgumentError("non-hex character")),
-            }
-        }
-        let s: &[u8] = s.as_ref();
-        if s.len() % 2 != 0 {
-            return Err(ClientError::ArgumentError("invalid length"));
-        }
-        let mut buf = vec![0; s.len() / 2];
-        for i in 0..buf.len() {
-            buf[i] = (v(s[i * 2])? << 4) | v(s[i * 2 + 1])?;
-        }
-        Ok(Self(buf))
-    }
-}
 
 impl AsRef<[u8]> for HexArg {
     fn as_ref(&self) -> &[u8] {
         &self.0
     }
 }
 
-#[derive(Debug, StructOpt)]
-#[structopt(
-    name = "neqo-client",
-    about = "A basic QUIC HTTP/0.9 and HTTP/3 client."
-)]
+fn decode_hex(s: &str) -> Result<Vec<u8>, hex::FromHexError> {
+    hex::decode(s)
+}
+
+#[derive(Debug, Parser)]
+#[command(author, version, about, long_about = None)]
 pub struct Args {
-    #[structopt(short = "a", long, default_value = "h3")]
+    #[arg(short = 'a', long, default_value = "h3")]
     /// ALPN labels to negotiate.
     ///
     /// This client still only does HTTP/3 no matter what the ALPN says.
     alpn: String,
 
     urls: Vec<Url>,
 
-    #[structopt(short = "m", default_value = "GET")]
+    #[arg(short = 'm', default_value = "GET")]
     method: String,
 
-    #[structopt(short = "h", long, number_of_values = 2)]
+    #[arg(short = 'H', long, number_of_values = 2)]
```

Comment:
```suggestion
    #[arg(short = 'h', long, number_of_values = 2)]
```

why use a capitalized letter here?

Reply:
Presumably because `-h` == `--help` usually.  And curl uses `-H` for headers, so that consistency is nice.

Reply:
What @martinthomson said